### DISCUSSION TITLE
ckb 0.103.0 (e77138e 2022-04-11)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#3373] [Announcements]
**Attention** that v0.103.0 will activate ckb2021 in mainnet since epoch 5414, at about 2022/05/10 1:00 UTC. See [RFC37](https://github.com/nervosnetwork/rfcs/blob/master/rfcs/0037-ckb2021/0037-ckb2021.md) and the [migration guide](https://github.com/jordanmack/nervos-ckb2021-hard-fork-migration-guide) for details.

Compatability Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2021 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2019 until epoch 5414 | v0.25.0             |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Downloads

| OS      | Arch  | Runtime Dependencies      | Package               | Sign               |
| ------- | ----- | ------------------------- | --------------------- | ------------------ |
| macOS   | x64   | macOS 10.15 or above      | [zip][macos-zip]      | [PGP][macos-pgp]   |
| Linux   | x64   | glibc, libstdc++          | [tarball][linux-zip]  | [PGP][linux-pgp]   |
| CentOS  | x64   | glibc, libstdc++, openssl | [tarball][centos-zip] | [PGP][centos-pgp]  |
| Windows | x64   | VC++ Redistributable      | [zip][windows-zip]    | [PGP][windows-pgp] |
| Linux   | ARM64 | glibc, libstdc++          | [tarball][arm-zip]    | [PGP][arm-pgp]     |
| Docker  |       |                           | [dockerhub][]         |                    |

> The binaries require following CPU instructions: call (MODE64), cmovbe (CMOV), xorps (SSE1), movq (SSE2)

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-apple-darwin.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-unknown-linux-gnu.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-unknown-centos-gnu.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.103.0/ckb_v0.103.0_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

<details><summary>Changes since v0.101.8</summary>

### Features

* #3371: Activate ckb2021 in mainnet since epoch 5414 (@doitian)
    At about 2022/05/10 1:00 UTC.

### Bug Fixes

* #3343: Genesis dep groups should supports relative file path (@quake)
* #3345: Fix(cli) add h256 validator for `--assume-valid-target` and `--ba-code-hash` (@chanhsu001)

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.103.0'>ckb 0.103.0 (e77138e 2022-04-11)</a>.</em>

**Comment by joshyates1980** [comment]
As I am not developing at this time on the Nervos Network, but only executing the CKB full node with version "ckb_v0.101.3_x86_64-pc-windows-msvc" when my computer is in operation, do you still recommend the 0.1.03.0 upgrade? The download of the full chain is big, long, and I want to make sure it is not necessary for my current situation (hence, not developing). Much gratitude, - Josh

**Reply by driftluo** [reply]
Upgrading is still recommended and it has several reasons:
1. 0.103.0 will perform hardfork and will reject connections from lower versions upon completion
2. 0.101.3 - 0.103.0 with some optimizations and fixes

The upgrade does not need to delete the historical data, just replace the ckb binary, that is, there is no need to resync all the data

**Reply by joshyates1980** [reply]
Thank you for the suggestion of "just replace the ckb binary." Great tip!
